File: src/main/java/com/github/codingvampyre/drachenfest/domain/Event.java
package com.github.codingvampyre.drachenfest.domain;

import lombok.*;

import javax.persistence.Entity;
import javax.persistence.OneToMany;
import javax.validation.constraints.NotNull;
import java.util.Date;
import java.util.List;
import java.util.UUID;

@Entity
@Getter
@Setter
@ToString
@RequiredArgsConstructor
@AllArgsConstructor
public class Event extends BaseEntity {
    @NotNull
    private String name;

    @NotNull
    private UUID uuid;

    @NotNull
    private Date beginAt;

    @NotNull
    private Date endAt;

    @OneToMany(mappedBy = "event")
    private List<TicketType> ticketTypes;
}

File: src/main/java/com/github/codingvampyre/drachenfest/domain/Event.kt
package com.github.codingvampyre.drachenfest.domain

import java.util.*
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.OneToMany

@Entity
open class Event(
		@Column(nullable = false)
		open var name: String? = null,

		@Column(nullable = false)
		open var uuid: UUID? = null,

		@Column(nullable = false)
		open var beginAt: Date? = null,

		@Column(nullable = false)
		open var endAt: Date? = null,

		@OneToMany(mappedBy = "event")
		open var ticketTypes: MutableList<TicketType> = mutableListOf(),
) : BaseEntity()
